Xamarin.PropertyEditing.Mac/Controls/PropertyEditorControl.cs

```diff
@@ -104,4 +107,13 @@ protected virtual void OnPropertyChanged (object sender, System.ComponentModel.P
 
 		protected abstract void UpdateAccessibilityValues ();
 	}
+
+	internal abstract class PropertyEditorControl<T> : PropertyEditorControl
```

Comment:
It'd be better to do something like:
```csharp
internal abstract class PropertyEditorControl<TViewModel>
    : PropertyEditorControl
    where TViewModel : PropertyViewModel
```

Some property types have viewmodels that subclass from the standard `PropertyViewModel<T>`, if we just let them declare the VM type they expect, they won't need to do this work _again_.

Reply:
Good point.

Reply:
@ermau What I was going to ping you about was.
1. Shouldn't the TViewModel above be of type PropertyViewModel<T>?
2. Are you suggesting this new class will be used across both Mac and Windows?
3. When you say "let them declare the VM type they expect" are you referring to when they register the mapping between VM and Type?

I think I'm misunderstanding exactly what you mean or possibly over complicating it in my head.

Reply:
1. It is: `where TViewModel : PropertyViewModel`
2. No.
3. No. The individual editor implementations can declare the VM they expect to be dealing with when they subclass `PropertyEditorControl<TViewModel>`. Ex: `class PredefinedValuesEditor<T> : PropertyEditorControl<PredefinedValuesViewModel<T>>`

Reply:
Still no, `ObjectPropertyViewModel` does not use the strongly typed version for example.